### PR TITLE
Enable sos's PrintException command

### PR DESCRIPTION
**Pull request by mikem8361**
Enabled the sos PrintException code to build and fixed a few problems. Some init code was needed to be enabled now that the global pointer table was implemented. 

Implemented PAL_IsDebuggerPresent (for Linux) so System.DebugBreak with stop in lldb. lldb has some problem stopping on SIGTRAP (int3) (at least for me).

### CODE REVIEW

**Comment by mikem8361** [comment]
@jkotas, @noahfalk could you review?

File: src/pal/src/init/pal.cpp [hunk 1]
```diff
@@ -653,8 +655,35 @@ BOOL
 PALAPI
 PAL_IsDebuggerPresent()
 {
-    // Always retun false for now.
+#ifdef __LINUX__
+    BOOL debugger_present = FALSE;
+    char buf[1024];
```
Comment (jkotas) [inline src/pal/src/init/pal.cpp]: The size of /proc/self/status is pretty close to 1k on my machine. Would it make sense to make the buffer size 2k to make this a bit future proof?

**Comment by jkotas** [comment]
LGTM modulo comment. Could you please squash the commits before merging?

**Comment by janvorli** [comment]
@mikem8361 It looks like the IsDebuggerPresent is called from quite many places and it seems that I can possibly happen quite frequently. I am worried a bit about performance consequences of reading the pseudofile and parsing its textual contents. Is there no more low level way of getting the information we need?

**Comment by mikem8361** [comment]
The other ways I found to detect a native debugger were even worse. They involved a fork(). We could do it once during init (CorHost->Start()?) and save it away.

http://stackoverflow.com/questions/3596781/detect-if-gdb-is-running

**Comment by mikem8361** [comment]
I realized doing the check during initialization won’t work if we want to support attaching after init.

mikem

From: Mike McLaughlin
Sent: Monday, March 16, 2015 2:46 PM
To: 'dotnet/coreclr'; dotnet/coreclr
Subject: RE: [coreclr] Enable sos's PrintException command (#466)

The other ways I found to detect a native debugger were even worse. They involved a fork(). We could do it once during init (CorHost->Start()?) and save it away.

http://stackoverflow.com/questions/3596781/detect-if-gdb-is-running

**Comment by jkotas** [comment]
I wonder whether we need to bother with detecting the debugger at all.  Would it make sense to keep this as is (unconditionally return false), and instead fix the places that do not work well with it instead (there should be a very few of them)?

**Comment by mikem8361** [comment]
I don’t understand. There are dozens of places that IsDebuggerPresent is called and all of these places need to know if there is a native debugger present.  Like in the System.Diagnostics.Debugger.DebugBreak case, are you suggesting removing the call to IsDebuggerPresent() and always try to execute the int3 (via DBG_DebugBreak)? Otherwise, if IsDebuggerPresent is always false like it is before my change, we will never stop in the native debugger.

mikem

**Comment by jkotas** [comment]
>>> removing the call to IsDebuggerPresent() and always try to execute the int3 (via DBG_DebugBreak)?
Yes, it is what I had in mind - via DebugBreak(), for FEATURE_PAL only.

**Pull request merged**